app/models/history.rb

```diff
@@ -39,6 +39,14 @@ class History < ApplicationRecord
 
   belongs_to :patient
 
+  # Patient updated_at should not be updated if history was created by a system generated action
```

Comment:
Might wanna clarify this comment - it's very specific cases we don't want to update the Patient record, not just any system generated action. The cases are 1) user collecting data about the monitoree and 2) the system still trying to send messages to the record

Reply:
I actually just realized something very silly about what we had suggested early when re-reading this. Simply preventing the history messages related to sending messages is of course not enough to handle this use case:
> 2) the system still trying to send messages to the record 

The system won't be sending messages to records that are closed anyway, and the close logic does not look at `updated_at`, only the purge logic does. So I don't actually think we even need to check the `report_reminder` and `contact_attempt` history types here. Even if a message was somehow sent to a closed record (like in the HoH case), `last_assessment_reminder_sent` would be updated on the Patient record so this check doesn't really do anything. 

So I'm basically proposing we change this check entirely to only filter out the `monitoree_data_downloaded` history type. @holmesie thoughts?

Reply:
The original statement:
>>ANY history item that is about sending the monitoree a message via the system automatically (whether it was a success or failure) should NOT be considered a record update.

Does not have anything to do with the state the Patient is in. I think the logic here is still valid - do not touch the `updated_at` of a patient when history items about system-generated contact attempts and data exports are created.

I'm not sure where the requirement for these specific cases came from but it would be against the above request to leave out these cases just because the features, in the way that they are currently implemented, do not take this into account.

Unless of course that the request wasn't discussing in general, but just as a way to have the "correct" Patient show up for purge eligible.

Reply:
>I'm not sure where the requirement for these specific cases came from but it would be against the above request to leave out these cases just because the features, in the way that they are currently implemented, do not take this into account.

Requirements came from discussion with PO team where we were revisiting the close/purge logic and this came up. 
We are essentially looking into updating likely the close logic to check for very stale records in the system - in which case we may be checking the `updated_at` field, and we didn't want that to be effected by automated messages or user downloads.

You're right in that we shouldn't be focused on the current close logic (because if we update the logic as I mentioned above that won't matter), so the more important part of what I said above is that when we do send messages, we already update an attribute on the Patient record (`last_assessment_reminder_sent`) so this change is rendered useless. I'm basically saying, because of this, we'd need another way to check for stale records other than looking at the `updated_at` anyway, so I don't think we need these extra checks.

Reply:
sounds good, so is the verdict to remove the `:report_reminder` and `:contact_attempt` edge cases and just keep the `:monitoree_data_download` one for the scope of this ticket and then update the purge logic in a separate ticket or try to do that here?

Reply:
Yes that's what I'm thinking. Would like @holmesie thought's before you make changes though.

Reply:
That makes sense to me as well @timwongj

Reply:
@kylietmo thanks for the additional explanation

Reply:
updated here: https://github.com/SaraAlert/SaraAlert/pull/683/commits/6a640aa26e785d5c839ca18884a3ee37181ea4ef